bindinfo/bind.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bindinfo
+
+import "github.com/pingcap/parser/ast"
+
+// BindHint will add hints for originStmt according to hintedStmt' hints.
+func BindHint(originStmt, hintedStmt ast.StmtNode) ast.StmtNode {
```

Comment:
Can we use the `visitor` design pattern to implement BindHint, like:
``` go
type BindHintHelper struct{
	hintedStmt ast.StmtNode
}

func (h *BindHintHelper) Enter(inNode ast.Node) (ast.Node, bool) {
	switch inNode.(type){
	case *ast.SelectStmt:
		return inNode, false
	default:
		return inNode, true
	}
}


func (h *BindHintHelper) Leave(inNode ast.Node) (ast.Node, bool) {
	switch x := inNode.(type) {
	case *ast.SelectStmt:
		return selectBind(x, h.hintedStmt.(*ast.SelectStmt)),true
	case *ast.ResultSetNode:

	}
}
```

Reply:
Enter will only return inNode, I think do this is not well helpful.